docs/ops/deployment/overview.md

```diff
@@ -0,0 +1,358 @@
+            <td>
+                <ul>
+                    <li><a href="">Kubernetes</a></li>
+                    <li><a href="">YARN</a></li>
+                    <li><a href="">Mesos</a></li>
+                    <li><a href="">Standalone</a> (this is the barebone mode that requires just JVMs to be launched. Deployment with Docker, Docker Swarm / Compose, non-native Kubernetes and other models is possible through manual setup in this mode)
+                    </li>
+                </ul>
+            </td>
+        </tr>
+        <tr>
+            <td colspan="3" class="text-center">
+                <b>External Components</b> (all optional)
+            </td>
+        </tr>
+        <tr>
+            <td>High Availability Service Provider</td>
+            <td>
+                Flink’s JobManager supports a high availability mode, where multiple JobManagers participate in a leader election process, resulting in an active JobManager and potentially multiple standby JobManagers, allowing for fast failover in case the active JobManager is lost.
+            </td>
+            <td>
+                <ul>
+                    <li><a href="">Zookeeper</a></li>
+                    <li><a href="">Kubernetes HA</a></li>
+                </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>File Storage and Persistency</td>
+            <td>
+                For checkpointing (recovery mechanism for streaming jobs) Flink relies on external file storage systems
+            </td>
+            <td>See <a href="">FileSystems</a> page.</td>
+        </tr>
+        <tr>
+            <td>Resource Provider</td>
+            <td>
+                Flink can be deployed through different Resource Provider Frameworks, such as Kubernetes, YARN or Mesos.
+            </td>
+            <td>See "JobManager" implementations above.</td>
+        </tr>
+        <tr>
+            <td>Metrics Storage</td>
+            <td>
+                Flink components report internal metrics and Flink jobs can report additional, job specific metrics as well.
+            </td>
+            <td>See <a href="">Metrics Reporter</a> page.</td>
+        </tr>
+        <tr>
+            <td>Application-level data sources and sinks</td>
+            <td>
+                While application-level data sources and sinks are not technically part of the deployment of Flink cluster components, they should be considered when planning a new Flink production deployment. Colocating frequently used data with Flink can have significant performance benefits
+            </td>
+            <td>
+                For example:
+                <ul>
+                    <li>Apache Kafka</li>
+                    <li>Amazon S3</li>
+                    <li>ElasticSearch</li>
+                    <li>Apache Cassandra</li>
+                </ul>
+                See <a href="">Connectors</a> page.
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+
+
+## Deployment Modes
```

Comment:
We should name these things in the same way where possible: The overview calls this the "job submission modes".